src/mpid/ch4/netmod/ofi/ofi_send.h

```diff
@@ -396,81 +369,27 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_OFI_send(const void *buf, MPI_Aint count, MPI
     MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_OFI_SEND);
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_OFI_SEND);
 
+    uint64_t cq_data = comm->rank;
+    MPIDI_OFI_idata_set_error_bits(&cq_data, err_flag);
+
     MPIDI_Datatype_get_info(count, datatype, dt_contig, data_sz, dt_ptr, dt_true_lb);
 
-    if (likely(!syncflag && dt_contig && (data_sz <= MPIDI_OFI_global.max_buffered_send)))
-        if (noreq)
-            mpi_errno =
-                MPIDI_OFI_send_lightweight((char *) buf + dt_true_lb, data_sz, comm->rank, dst_rank,
-                                           tag, comm, context_offset, addr);
-        else
-            mpi_errno = MPIDI_OFI_send_lightweight_request((char *) buf + dt_true_lb, data_sz,
-                                                           comm->rank, dst_rank, tag,
-                                                           comm, context_offset, addr, request);
-    else
-        mpi_errno = MPIDI_OFI_send_normal(buf, count, datatype, dst_rank, tag, comm,
+    if (likely(!syncflag && dt_contig && (data_sz <= MPIDI_OFI_global.max_buffered_send))) {
+        mpi_errno = MPIDI_OFI_send_lightweight((char *) buf + dt_true_lb, data_sz,
+                                               cq_data, dst_rank, tag, comm, context_offset, addr);
+        if (!noreq) {
+            *request = MPIR_Request_create_complete(MPIR_REQUEST_KIND__SEND);
+        }
```

Comment:
NOTE: @hzhou if `*request` is passed down (from workq?) then we need complete the request here -- just need add an `else` branch, right?